docs/checkpoints.md

```diff
@@ -0,0 +1,26 @@
+# State checkpoints
+If enabled in the config, every _epoch length * checkpoint interval_ blocks, the node will export a state checkpoint in its datadir. This snapshot can then be imported into a brand new node to serve as a checkpoint from which to start syncing (rather than using genesis), fast-tracking the sync.
+
+## Configuration
+ * `do_checkpoints`, *boolean (default `false`)*: If `true`, enables exporting the checkpoint files. If false, checkpoints will not be created.
+ * `consensus.epochs_per_checkpoint`, *u64 (default 24)*: If `do_checkpoints` is true, determines the frequency (in epochs) at which checkpoints will be exported. This is a consensus property and will eventually be obtained on-chain rather than being configurable.
+ * `load_checkpoint`, *struct, optional (default `None`)*: If provided, on startup, the node will attempt to load the given checkpoint and start syncing from it. Fails if the node has a non-empty database, i.e. intended to be used only for brand new nodes without existing data.
+   * `load_checkpoint.file`, *filepath*: The file to read the checkpoint from
+   * `load_checkpoint.hash`, *string*: The hex-encoded hash of the checkpoint block. This value is assumed to be trusted; the checkpoint contents are verified against this hash.
+
+## Directory
+Checkpoint files are saved inside the node's data directory, at the path `/checkpoints/{block_height}`.
+
+If the node does not have a data directory (i.e. is running on an ephemeral in-memory database), no checkpoints will be exported, regardless of the `do_checkpoints` parameter.
+
+Currently, old checkpoints are _not_ kept. Once a new checkpoint file is successfully created, any other files in the directory are deleted.
```

Comment:
What motivated this choice? Given we are deferring responsibility for uploading/publishing the checkpoints to external tooling, I'd argue we should defer the responsibility of cleanup to the same tool. Otherwise, we run the possibility of the node deleting checkpoints from underneath the tool trying to upload them.